pkg/planner/core/preprocess.go

```diff
@@ -294,6 +326,42 @@ func (p *preprocessor) extractSchema(in ast.Node) []pmodel.CIStr {
 		dbNames = append(dbNames, node.Table.Schema)
 	case *ast.LoadDataStmt:
 		dbNames = append(dbNames, node.Table.Schema)
+	case *ast.InsertStmt:
+		for _, tbl := range p.resolveCtx.GetTableNames() {
```

Comment:
will we also forbid `insert into non_readonly select * from readonly`?

Reply:
Yes, you are right, and fix this in https://github.com/pingcap/tidb/pull/62985/commits/578c2241786365ee7e4973e952b2dac47a58d79c